src/uu/test/src/test.rs

```diff
@@ -12,10 +12,24 @@ mod parser;
 
 use parser::{parse, Symbol};
 use std::ffi::{OsStr, OsString};
-
-pub fn uumain(args: impl uucore::Args) -> i32 {
-    // TODO: handle being called as `[`
-    let args: Vec<_> = args.skip(1).collect();
+use std::path::Path;
+
+pub fn uumain(mut args: impl uucore::Args) -> i32 {
+    let program = args.next().unwrap_or_else(|| OsString::from("test"));
+    let binary_name = Path::new(&program)
+        .file_name()
+        .unwrap_or_else(|| OsStr::new("test"))
+        .to_string_lossy();
+    let mut args: Vec<_> = args.collect();
+
+    // If invoked via name '[', matching ']' must be in the last arg
+    if binary_name == "[" {
+        let last = args.pop();
+        if last != Some(OsString::from("]")) {
+            eprintln!("[: missing `]'");
```

Comment:
To match GNU's output:
```suggestion
            eprintln!("[: ']' expected'");
```

Reply:
I don't get that, is that perhaps output from the shell builtin?

(Now I do notice I took the output from bash builtin, GNU is producing it with unicode quotes: `[: missing ‘]’`)

Reply:
The quotes are part of the locale. We're trying to stick with the C locale while we don't have proper locale support across all utils. You can try
```
LC_ALL=C env [
```
(the `env` ensures the GNU version is used)

That gives
```
[: missing ']'
```
on my machine (GNU coreutils 8.32)

Reply:
I think sylvestre's error message is the message given by zsh?

Reply:
oh yeah, sorry

Reply:
Switched to the LC_ALL=C version

Reply:
to reproduce
```
$ LANG=C /bin/\[ 1 -eq 1
/bin/[: missing ']'

```